framework/src/executioners/Eigenvalue.C

```diff
@@ -61,40 +78,133 @@ Eigenvalue::validParams()
 }
 
 Eigenvalue::Eigenvalue(const InputParameters & parameters)
-  : Steady(parameters),
+  : Executioner(parameters),
     _eigen_problem(*getCheckedPointerParam<EigenProblem *>(
         "_eigen_problem", "This might happen if you don't have a mesh")),
     _normalization(isParamValid("normalization") ? &getPostprocessorValue("normalization")
-                                                 : nullptr)
+                                                 : nullptr),
+    _system_time(getParam<Real>("time")),
+    _time_step(_eigen_problem.timeStep()),
+    _time(_eigen_problem.time()),
+    _final_timer(registerTimedSection("final", 1))
 {
 // Extract and store SLEPc options
 #if LIBMESH_HAVE_SLEPC
-  Moose::SlepcSupport::storeSlepcOptions(_fe_problem, parameters);
+  Moose::SlepcSupport::storeSlepcOptions(_eigen_problem, parameters);
 
   Moose::SlepcSupport::storeSlepcEigenProblemOptions(_eigen_problem, parameters);
   _eigen_problem.setEigenproblemType(_eigen_problem.solverParams()._eigen_problem_type);
-#endif
 
-  if (!parameters.isParamValid("normalization") && parameters.isParamSetByUser("normal_factor"))
+  // If need to initialize eigen vector
+  _eigen_problem.needInitializeEigenVector(getParam<bool>("auto_initialization"));
+
+  // Whether or not the system outputs the inverse of eigenvaue.
+  // It is useful for neutron calculations. The inverse of the eigenvalue is the multiplication
+  // factor.
+  _eigen_problem.outputInverseEigenvalue(getParam<bool>("output_inverse_eigenvalue"));
+
+  if (!isParamValid("normalization") && isParamValid("normal_factor"))
     paramError("normal_factor",
                "Cannot set scaling factor without defining normalization postprocessor.");
+
+  // _feproblem_solve calls FEProblemBase
+  _picard_solve.setInnerSolve(_feproblem_solve);
+  _time = _system_time;
+
+  if (isParamValid("normalization"))
+  {
+    auto normpp = getParam<PostprocessorName>("normalization");
```

Comment:
Not important but could be `const auto &` to avoid copies. I know that string is not very expensive to copy :smile:

Reply:
Haha. There is a lesson for me, I should make everything constant. And then git rid of these that can not be made as constant